docs/madr/decisions/083-meshservice-meshidentity-identity.md

```diff
@@ -0,0 +1,298 @@
+```
+These translate into Envoy configuration as follows:
+Envoy Cluster for Server (expects the Server’s identity):
+```yaml
+              matchTypedSubjectAltNames:
+              - matcher:
+                  exact: spiffe://default/server_default_kuma-demo_svc_80
+                sanType: URI
+```
+
+Envoy Cluster for Client (expects the Client’s identity):
+```yaml
+              matchTypedSubjectAltNames:
+              - matcher:
+                  exact: spiffe://default/client_default_kuma-demo_svc_80
+                sanType: URI
+```
+10. Traffic breaks because the Server presents itself as:
+
+`spiffe://default.zone.mesh.local/ns/kuma-demo/sa/server` 
+but the Client still expects:
+`spiffe://default/server_default_kuma-demo_svc_80`
+
+11. After the reconciliation loop (default 5s), the `MeshService` resources are updated:
+MeshService of Client
+```yaml
+spec:
+  identities:
+  - type: ServiceTag
+    value: client_default_kuma-demo_svc_80
+  - type: SpiffeID
+    value: spiffe://default.zone.mesh.local/ns/kuma-demo/sa/client
+```
+
+MeshService of Server
+```yaml
+spec:
+  identities:
+  - type: ServiceTag
+    value: server_default_kuma-demo_svc_80
+  - type: SpiffeID
+    value: spiffe://default.zone.mesh.local/ns/kuma-demo/sa/server
+```
+12. Envoy configuration is recalculated to accept both ServiceTag and SpiffeID identities:
+Envoy Cluster for Server (accepts Client identities):
+```yaml
+              matchTypedSubjectAltNames:
+              - matcher:
+                  exact: spiffe://default/server_default_kuma-demo_svc_80
+                sanType: URI
+              - matcher:
+                  exact: spiffe://default.zone.mesh.local/ns/kuma-demo/sa/client
+                sanType: URI
+```
+
+Envoy Cluster for Client (accepts Server identities):
+```yaml
+              matchTypedSubjectAltNames:
+              - matcher:
+                  exact: spiffe://default/client_default_kuma-demo_svc_80
+                sanType: URI
+              - matcher:
+                  exact: spiffe://default.zone.mesh.local/ns/kuma-demo/sa/server
+                sanType: URI
+```
+13. Traffic resumes successfully
+
+As we can see from this flow diagram, there is a period during which traffic from the new identity is not accepted.
+
+## User stories
+
+### As a Mesh Operator, I want to migrate from legacy mTLS to MeshIdentity without any traffic disruption
+
+This is a real use case: we want to migrate users to a new identity and avoid any problems with traffic during the process.
+
+### As a Mesh Operator, I want to migrate from one trust domain to another without downtime
+
+There might be cases where a user wants to migrate from one trust domain to another, and there should be no problems with traffic during this time.
+
+## Considered Options
+
+- Introduce a new field `mode`
+- Use a `kuma.io/effect` label
+
+## Decision Outcome
+
+- Introduce a new field `mode`
+
+## Design
+
+### Introduce a new field `mode`
```

Comment:
What if instead of new field `mode` we just allow creating MeshIdentity without the `provider`:
```yaml
apiVersion: kuma.io/v1alpha1
kind: MeshIdentity
metadata:
  name: identity
  namespace: kuma-system
  labels:
    kuma.io/mesh: default
spec:
  selector:
    dataplane:
      matchLabels: {}
  spiffeID:
    trustDomain: "{{ .Mesh }}.{{ .Zone }}.mesh.local"
    path: "/ns/{{ .Namespace }}/sa/{{ .ServiceAccount }}"
```
In this case spiffeID is propagated to MeshService, but no actual crypto identity is propagated to the DPP (because there is no provider).